proteuslib/unit_models/tests/test_reverse_osmosis_1D.py

```diff
@@ -293,4 +565,843 @@ def test_solution(self, RO_frame):
                 value(m.fs.unit.permeate_out[0].flow_mass_phase_comp['Liq', 'H2O']))
         assert (pytest.approx(1.8186e-4, rel=1e-3) ==
                 value(m.fs.unit.permeate_out[0].flow_mass_phase_comp['Liq', 'NaCl']))
-        assert (pytest.approx(0, rel=1e-3) == value(m.fs.unit.deltaP[0, .5]))
+
+    @pytest.mark.component
+    def TestReverseOsmosis_cp_mod_fixed(self):
+        m = ConcreteModel()
+        m.fs = FlowsheetBlock(default={"dynamic": False})
+
+        m.fs.properties = props.NaClParameterBlock()
+
+        m.fs.unit = ReverseOsmosis1D(default={
+            "property_package": m.fs.properties,
+            "has_pressure_change": False,
+            "concentration_polarization_type": ConcentrationPolarizationType.fixed,
+            "mass_transfer_coefficient": MassTransferCoefficient.none,
+            })
+
+        # fully specify system
+        feed_flow_mass = 1
+        feed_mass_frac_NaCl = 0.035
+        feed_pressure = 50e5
+        feed_temperature = 273.15 + 25
+        A = 4.2e-12
+        B = 3.5e-8
+        pressure_atmospheric = 101325
+        feed_mass_frac_H2O = 1 - feed_mass_frac_NaCl
+
+        m.fs.unit.feed_inlet.flow_mass_phase_comp[0, 'Liq', 'NaCl'].fix(
+            feed_flow_mass * feed_mass_frac_NaCl)
+
+        m.fs.unit.feed_inlet.flow_mass_phase_comp[0, 'Liq', 'H2O'].fix(
+            feed_flow_mass * feed_mass_frac_H2O)
+
+        m.fs.unit.feed_inlet.pressure[0].fix(feed_pressure)
+        m.fs.unit.feed_inlet.temperature[0].fix(feed_temperature)
+        m.fs.unit.A_comp.fix(A)
+        m.fs.unit.B_comp.fix(B)
+        m.fs.unit.permeate_outlet.pressure[0].fix(pressure_atmospheric)
+        m.fs.unit.length.fix(8)
+        m.fs.unit.recovery_vol_phase[0, 'Liq'].fix(0.4)
+        m.fs.unit.cp_modulus.fix(1.1)
+
+        # test ports
+        port_lst = ['feed_inlet', 'feed_outlet', 'permeate_outlet']
+        for port_str in port_lst:
+            port = getattr(m.fs.unit, port_str)
+            assert len(port.vars) == 3  # number of state variables for NaCl property package
+            assert isinstance(port, Port)
+
+        # test pyomo objects on unit
```

Comment:
We don't need to do this for each case, I'd get rid of this for all but the first one.

Reply:
I decided to do this since different cases can have a slightly different set of variable/constraint objects. So while this may seem exhaustive, I suppose it does serve a purpose.

Reply:
Yeah but then I would just check the things that are different from the main detailed case. Regardless all of this will be reduced with a unit test harness that hopefully I do next week.